datalake/src/main/java/com/sequenceiq/datalake/flow/detach/event/SdxStartDetachRecoveryEvent.java

```diff
@@ -0,0 +1,47 @@
+package com.sequenceiq.datalake.flow.detach.event;
+
+import java.util.Objects;
+
+import com.sequenceiq.cloudbreak.common.event.AcceptResult;
+import com.sequenceiq.datalake.flow.SdxEvent;
+import com.sequenceiq.datalake.flow.detach.SdxDetachEvent;
+
+import reactor.rx.Promise;
+
+public class SdxStartDetachRecoveryEvent extends SdxEvent {
+    private final Exception exception;
+
+    public SdxStartDetachRecoveryEvent(String selector, Long detachedSdxId, String userId) {
+        super(selector, detachedSdxId, userId);
+        exception = null;
+    }
+
+    public SdxStartDetachRecoveryEvent(String selector, Long detachedSdxId, String userId, Promise<AcceptResult> accepted) {
+        super(selector, detachedSdxId, userId, accepted);
+        exception = null;
+    }
+
+    public SdxStartDetachRecoveryEvent(Long detachedSdxId, String userId, Exception exception) {
+        super(SdxDetachEvent.SDX_DETACH_RECOVERY_EVENT.event(), detachedSdxId, userId);
+        this.exception = exception;
+    }
+
+    public Exception getException() {
+        return exception;
+    }
+
+    public boolean isFailureEvent() {
+        return exception != null;
+    }
```

Comment:
this is a bit odd, and it I think the logic could be simplified/be more obvious if there was a different action/state for the failure with a specific selector/event for that case

Reply:
While I understand your rationale here, I personally don't know how simple it would be to introduce an entirely different event directed to the same state as this event.
I also don't want to create a new action/state since it would contain the exact logic as the state here. They only differ in their final transitions.

I am open to making this a little cleaner, such as having a tracked boolean instead or some other logic, but I personally don't think adding more events and actions would keep this simple and understandable.

Reply:
Okay I'll get this changed today, while hopefully reusing events and the like so as to not make this too complex. I'll retest everything again too.